Move defconfig and device tree from meta-xt-rcar to product layer

iusyk opened the pull request.

lorc commented:
Could you please fix the commit descriptions?

We don't allow lines longer than 80 characters for the commit messages, and no more than 70 for the commit captions

lorc commented on meta-xt-prod-devel-rcar-driver-domain/recipes-kernel/linux/linux-renesas_%.bbappend:
```diff
@@ -0,0 +1,4 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+
+ADDITIONAL_DEVICE_TREES = "${XT_DEVICE_TREES}"
```
I don't think you need move this variable from the original file

iusyk commented:
> I fully support idea to relocate *.dts but I see that incorrect DTS is used for KF (`*-kf-dom?.dts`). Need to take into account changes from [xen-troops/meta-xt-rcar#19](https://github.com/xen-troops/meta-xt-rcar/pull/19). Explanation is provided in PR.

fixed and rebased

rshym commented:
Wrong content for *-kf-domd.dts - it's not DTS, it's domd.cfg.
Wrong content for *-kf-xen.dts - it is taken from REL7.0 which doesn't work for KF. We need to take it from prod-devel branch master.

iusyk commented:
> 

it has been built and tested at KF

iusyk commented:
> 

regarding *-kf-xen.dts - it has been taken from meta-xt-rcar (branch master)

rshym commented on meta-xt-prod-devel-rcar-control/recipes-guests/domd/files/domd-h3ulcb-4x2g-kf.cfg:
```diff
@@ -1,44 +1,43 @@
     "/audio_clk_b",
     "/audio_clk_c",
     "/can",
+    "/opp_table0",
```
Please take domd-h3ulcb-4x2g-kf.cfg from old prod-devel master HEAD. There are no nodes like `opp_tableX` or `firmware`. And mentioned config (from old prod-devel master) is proven to work on new prod-devel-rcar KF.

rshym commented on meta-xt-prod-devel-rcar-control/recipes-guests/domd/files/domu-generic-h3-4x2g.cfg:
```diff
@@ -0,0 +1,55 @@
+# =====================================================================
+# DomU guest configuration
```
Please remove this domU.cfg from domD folder.
DomU config is provided by DomU target.

rshym commented on prod-devel-rcar.yaml:
```diff
@@ -7,6 +7,7 @@ variables:
   DOMD_BUILD_DIR: "build-domd"
   DOMU_BUILD_DIR: "build-domu"
   XT_DOMD_DTB_NAME: "%{SOC_FAMILY}-%{MACHINE}-domd.dtb"
+  XT_DOMUSER_DTB_NAME: ""
```
I checked in sources that DomU's DTB is provided to dom0 by 
```
install -m 0644 ${S}/${XT_DOMU_DTB_NAME} ${D}${libdir}/xen/boot/domu.dtb
```
in https://github.com/xen-troops/meta-xt-common/blob/master/meta-xt-control-domain/recipes-guest/domu/domu.bb#L29
So I'm not sure we need this commit. Also I have tested that DomU successfully runs on prod-devel-rcar without this change (with fixed bl2 and proper domd.cfg).

But we really need to clarify and describe in documentation, which device tree should be provided by the domain's recipes and which have to be added to `XT_DEVICE_TREES`. But this need to be discussed.

rshym commented on meta-xt-prod-devel-rcar-driver-domain/recipes-kernel/linux/files/r8a77951-h3ulcb-4x2g-ab-domd.dts:
```diff
@@ -0,0 +1,82 @@
+ */
+
+#include "r8a77951-ulcb.dts"
+#include "ulcb-ab.dtsi"
```
This file `ulcb-ab.dtsi` is not moved from meta-xt-rcar.

rshym commented:
I was able to run prod-devel-rcar domD+domU on KF with first two commits and fixed domd.cfg.
So, please:
- remove last commit (domu.dts)
- fix domd.cfg related comit (replace domd.cfg with proper one and remove domu.cfg)
- add ulcb-ab.dtsi to first commit

After that, (and removal or ulcb-ab.dtsi from xt-rcar #23) we are ready to merge.

rshym commented:
`Reviewed-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>`

Please merge it with https://github.com/xen-troops/meta-xt-rcar/pull/23
